archive_extractor/limiting_aggregating_read_closer.go

```diff
@@ -2,11 +2,29 @@ package archive_extractor
 
 import (
 	"errors"
+	"fmt"
 	"io"
 )
 
 var ErrTooManyEntries = errors.New("too many entries in archive")
-var ErrCompressLimitReached = errors.New("total bytes limit reached")
+
+type ErrCompressLimitReached struct {
+	SizeLimit int64
+	CurrSize  int64
+}
+
+func newErrCompressLimitReached(sizeLimit, total int64) *ErrCompressLimitReached {
+	return &ErrCompressLimitReached{SizeLimit: sizeLimit, CurrSize: total}
+}
+
+func IsErrCompressLimitReached(err error) bool {
+	_, ok := err.(*ErrCompressLimitReached)
+	return ok
+}
+
+func (ErrCompressLimit *ErrCompressLimitReached) Error() string {
+	return fmt.Sprintf("total bytes limit reached with the following values: size limit: %d, total current size: %d", ErrCompressLimit.SizeLimit, ErrCompressLimit.CurrSize)
```

Comment:
Start with a Capital letter

Reply:
The standard in go is lower case for error strings